src/Platform/Microsoft.Testing.Platform/OutputDevice/Console/ConsoleLogger.cs

```diff
@@ -0,0 +1,702 @@
+                asm.PassedTests++;
+                asm.TotalTests++;
+                break;
+            case LoggerOutcome.Skipped:
+                asm.SkippedTests++;
+                asm.TotalTests++;
+                break;
+            case LoggerOutcome.Timeout:
+                asm.TimedOutTests++;
+                asm.TotalTests++;
+                break;
+            case LoggerOutcome.Cancelled:
+                asm.CancelledTests++;
+                asm.TotalTests++;
+                break;
+        }
+
+        var update = new TestWorker(asm.PassedTests, asm.FailedTests, asm.SkippedTests, Path.GetFileName(asm.Assembly), asm.TargetFramework, asm.Architecture, asm.Stopwatch, null);
+        _consoleWithProgress.UpdateWorker(asm.SlotIndex, update);
+        if (outcome != LoggerOutcome.Passed || _options.ShowPassedTests)
+        {
+            _consoleWithProgress.WriteToTerminal(terminal => RenderTestCompleted(
+                terminal,
+                assembly,
+                targetFramework,
+                architecture,
+                displayName,
+                outcome,
+                duration,
+                errorMessage,
+                errorStackTrace,
+                expected,
+                actual));
+        }
+    }
+
+    internal /* for testing */ void RenderTestCompleted(
+        ITerminal terminal,
+        string assembly,
+        string? targetFramework,
+        string? architecture,
+        string displayName,
+        LoggerOutcome outcome,
+        TimeSpan duration,
+        string? errorMessage,
+        string? errorStackTrace,
+        string? expected,
+        string? actual)
+    {
+        if (outcome == LoggerOutcome.Passed && !_options.ShowPassedTests)
+        {
+            return;
+        }
+
+        TerminalColor color = outcome switch
+        {
+            LoggerOutcome.Error or LoggerOutcome.Fail or LoggerOutcome.Cancelled or LoggerOutcome.Timeout => TerminalColor.DarkRed,
+            LoggerOutcome.Skipped => TerminalColor.DarkYellow,
+            LoggerOutcome.Passed => TerminalColor.DarkGreen,
+            _ => throw new NotImplementedException(),
+        };
+        string outcomeText = outcome switch
+        {
+            LoggerOutcome.Fail or LoggerOutcome.Error => PlatformResources.FailedLowercase,
+            LoggerOutcome.Skipped => PlatformResources.SkippedLowercase,
+            LoggerOutcome.Cancelled or LoggerOutcome.Timeout => $"{PlatformResources.FailedLowercase} ({PlatformResources.CancelledLowercase})",
+            LoggerOutcome.Passed => PlatformResources.PassedLowercase,
+            _ => throw new NotImplementedException(),
+        };
+
+        terminal.SetColor(color);
+        terminal.Append(outcomeText);
+        terminal.ResetColor();
+        terminal.Append(' ');
+        terminal.Append(displayName);
+        terminal.SetColor(TerminalColor.Gray);
+        terminal.Append(' ');
+        terminal.Append($"({duration.TotalSeconds:F1}s)");
```

Comment:
We may want to reuse the helper here (that takes care of ms, min...)

Reply:
done.